app/frontend/src/features/profile/hooks/referencesHooks.ts

```diff
@@ -93,6 +94,7 @@ export function useWriteHostReference(userId: number) {
     {
       onSuccess: () => {
         queryClient.invalidateQueries([availableWriteReferencesKey(userId)]);
+        queryClient.invalidateQueries([referencesReceivedKey(userId, "all")]);
```

Comment:
```suggestion
        queryClient.invalidateQueries(referencesReceivedBaseKey);
```
Ahh my bad not noticing this earlier - but depending on if the reference is written to the person is a surfer or host reference, we'd have to invalidate that reference type too in addition to `all`.

But instead of doing that individually, I think it'd be easier if we update `referencesReceivedKey` to the following in `src/queryKeys.ts`:
```typescript
export const referencesReceivedBaseKey = "referencesReceived";
export const referencesReceivedKey = (
  userId: number,
  type: ReferenceType | "all"
) => [referenceReceivedBaseKey, { type, userId }];
```

So then you can use my suggestion to invalidate all of the queries that starts with `referencesReceived` without having to do a conditional check to see what type of reference they give etc and complicates the code, as I don't think it matters too much that we over invalidate in this case since the refetching will happen in the background anyway. Hope that makes sense!

Reply:
It's more my bad for not knowing! Still wrapping my head around all the `reactQuery` stuff, so I really appreciate the explanation 🙂 Just to confirm that my understanding is correct: 

> we'd have to invalidate that reference type too in addition to `all`

Is this is because when a user filters the References by reference type, there is an entirely new query done? So `all` does not cover all types of references, but only the one query when a user is specifically viewing references of all types – and selecting a new reference type (i.e. hosted) does a new query all together. So only invalidating `all` means when filtering by i.e. `hosted` that could potentially be a stale query. 

It just wasn't fully clicking before but looking at the code now I think I finally understand better.

Reply:
A question though: doesn't this then invalidate the queries for all `userId`s? I assume that's what you're referring to by: 
> I don't think it matters too much that we over invalidate in this case since the refetching will happen in the background anyway.

But would it not be preferable to invalidate the queries for that `userId` and all relevant types: `all`, `hosted`, and `surfed` for `writeHostRequestReference`, and `all` and `friend` for `writeFriendReference`?

Reply:
I did it the way you suggested in case the suggestion my previous comment isn't better, to have it ready to go.

Reply:
> Is this is because when a user filters the References by reference type, there is an entirely new query done?

Yes, when the user changes that filter to say only show references from hosts, the query key used for that becomes ["referencesReceived", { userId: 3, type: ReferenceType.REFERENCE_TYPE_HOSTED }]`, and so `all` won't cover this query. It was done this way so we delegate the heavy lifting of filtering to the backend, unlike before where I had to do the same work in the frontend before, quite similar to the work you did in a different PR before I ask the backend team to update their API

> doesn't this then invalidate the queries for all userIds?

That's a very good point, I overlooked this when I said it doesn't matter if we over invalidate as I was only thinking along the query types but not the user ids 😅  so you are right, I think we'd be invalidating too many queries.

In that case, I think we'll have to use the `predicate` option of `invalidateQueries` and do something like this:
```typescript
queryClient.invalidateQueries({
   predicate: ({ queryKey }) =>
     queryKey[0] === referencesReceivedBaseKey && queryKey[1]?.userId === userId,
 })
```

Reply:
Ok, I had to do some type assertions to get it to work. Hopefully this does it, let me know 🤞

Reply:
I had a look over and the types are pretty crap/generic, so I think it's ok as an escape hatch for this case to type assert :)